src/lib.rs

```diff
@@ -294,7 +294,10 @@ pub struct Package {
     pub authors: Vec<String>,
     /// An opaque identifier for a package
     pub id: PackageId,
-    source: Option<String>,
+    /// The source of the package, e.g. 
+    /// "registry+https://github.com/rust-lang/crates.io-index"
+    /// or `None` for local projects.
```

Comment:
This may be more precisely stated that it is None for path dependencies or workspace members.

Reply:
Makes sesne!